backend/burning-okr/burning-okr-app/src/test/java/org/burningokr/config/WebSocketAuthenticationTest.java

```diff
@@ -0,0 +1,234 @@
+
+    //assert
+    Assertions.assertEquals("Authorization token in header is malformed. Expected structure: Bearer {token}", exc.getMessage());
+    verifyNoMoreInteractions(this.authenticationUserContextServiceMock, this.jwtAuthenticationConverterMock);
+  }
+
+  @Test
+  void tryToAuthenticate_shouldThrowAuthorizationHeaderException_whenIsNotAuthenticated() {
+    //assemble
+    StompHeaderAccessor stompHeaderAccessorMock = mock(StompHeaderAccessor.class);
+    doReturn(true).when(stompHeaderAccessorMock).containsNativeHeader(WebSocketAuthentication.WEBSOCKET_STOMP_HEADER_AUTHORIZATION_KEY);
+    doReturn(List.of("Bearer dummyBearerTokenString")).when(stompHeaderAccessorMock).getNativeHeader(WebSocketAuthentication.WEBSOCKET_STOMP_HEADER_AUTHORIZATION_KEY);
+
+    Jwt jwtMock = mock(Jwt.class);
+    doReturn(jwtMock).when(this.jwtDecoderMock).decode("dummyBearerTokenString");
+
+    AbstractAuthenticationToken abstractAuthenticationTokenMock = mock(AbstractOAuth2TokenAuthenticationToken.class);
+    doReturn(false).when(abstractAuthenticationTokenMock).isAuthenticated();
+    doReturn(abstractAuthenticationTokenMock).when(this.jwtAuthenticationConverterMock).convert(jwtMock);
+
+    User userMock = mock(User.class);
+    doReturn(userMock).when(this.authenticationUserContextServiceMock).getUserFromToken(jwtMock);
+
+    //act + assert
+    AuthorizationHeaderException exc = Assertions.assertThrows(AuthorizationHeaderException.class, () ->
+      this.webSocketAuthentication.tryToAuthenticate(stompHeaderAccessorMock));
+
+    //assert
+    Assertions.assertEquals("User is not authenticated", exc.getMessage());
+    verifyNoMoreInteractions(this.authenticationUserContextServiceMock, this.jwtAuthenticationConverterMock);
+  }
+
+  @Test
+  void tryToAuthenticate_shouldThrowRuntimeException_whenHeaderSessionAttributesAreNull() {
+    //assemble
+    StompHeaderAccessor stompHeaderAccessorMock = mock(StompHeaderAccessor.class);
+    doReturn(true).when(stompHeaderAccessorMock).containsNativeHeader(WebSocketAuthentication.WEBSOCKET_STOMP_HEADER_AUTHORIZATION_KEY);
+    doReturn(List.of("Bearer dummyBearerTokenString")).when(stompHeaderAccessorMock).getNativeHeader(WebSocketAuthentication.WEBSOCKET_STOMP_HEADER_AUTHORIZATION_KEY);
+
+    Jwt jwtMock = mock(Jwt.class);
+    doReturn(jwtMock).when(this.jwtDecoderMock).decode("dummyBearerTokenString");
+
+    AbstractAuthenticationToken abstractAuthenticationTokenMock = mock(AbstractOAuth2TokenAuthenticationToken.class);
+    doReturn(true).when(abstractAuthenticationTokenMock).isAuthenticated();
+    doReturn(abstractAuthenticationTokenMock).when(this.jwtAuthenticationConverterMock).convert(jwtMock);
+
+    User userMock = mock(User.class);
+    doReturn(userMock).when(this.authenticationUserContextServiceMock).getUserFromToken(jwtMock);
+
+    doNothing().when(this.authenticationUserContextServiceMock).updateCachedAndDatabaseUser(userMock);
```

Comment:
Should do nothing by default. This should only be needed when using a spy instead of a mock.

Reply:
That is absolutely correct! :)